src/entries/popup/pages/send/ToAddressInput.tsx

```diff
@@ -161,15 +162,45 @@ const DropdownWalletsList = ({
   watchedWallets: Address[];
   selectWalletAndCloseDropdown: (address: Address) => void;
 }) => {
-  const { walletOrder } = useWalletOrderStore();
+  const [newWalletOrder, setNewWalletOrder] = useState<`0x${string}`[]>([]);
+  const { walletOrder, saveWalletOrder } = useWalletOrderStore();
+
+  // find the items in getAccounts then compare to the current
+  // walletOrder, then return all items in getAccounts in order
+
+  useEffect(() => {
+    const combineAndOrderArrays = async (orderArray: string[]) => {
+      const orderSet = new Set(orderArray);
+      const primaryArray = await getAccounts();
+      const itemsNotInOrderArray = primaryArray.filter(
+        (item) => !orderSet.has(item),
+      );
+
+      const primaryMap = new Map(primaryArray.map((item) => [item, item]));
+
+      const combinedAndOrdered = [
+        ...orderArray.filter((item) => primaryMap.has(item)),
+        ...itemsNotInOrderArray,
+      ];
+
+      return combinedAndOrdered;
+    };
+
+    combineAndOrderArrays(walletOrder).then((orderedAccounts) => {
+      setNewWalletOrder(orderedAccounts as `0x${string}`[]);
+      saveWalletOrder(orderedAccounts as `0x${string}`[]);
+    });
+  }, [walletOrder, saveWalletOrder]);
+
   const sortedWallets = useMemo(
-    () => sortWallets(walletOrder, wallets),
-    [wallets, walletOrder],
+    () => sortWallets(newWalletOrder, wallets),
+    [newWalletOrder, wallets],
   );
   const sortedWatchedWallets = useMemo(
-    () => sortWallets(walletOrder, watchedWallets),
-    [watchedWallets, walletOrder],
+    () => sortWallets(newWalletOrder, watchedWallets),
+    [newWalletOrder, watchedWallets],
   );
```

Comment:
after playing with it a bit I'm pretty sure the props are in the right state already, we just need to remove all sorting from inside this component
like 
<img width="910" alt="Screenshot 2024-03-07 at 15 46 33" src="https://github.com/rainbow-me/browser-extension/assets/6232729/d563ce84-9824-49a8-88d3-052e4bd61da9">

Reply:
<img width="100" src="https://github.com/rainbow-me/browser-extension/assets/41711440/b162799a-133e-438b-8771-e83faf2d0feb" />

Reply:
changed lol